bitnami/mariadb-galera/templates/statefulset.yaml

```diff
@@ -0,0 +1,292 @@
+apiVersion: apps/v1beta1
+kind: StatefulSet
+metadata:
+  name: {{ template "mariadb-galera.fullname" . }}
+  labels:
+    app.kubernetes.io/name: {{ include "mariadb-galera.name" . }}
+    helm.sh/chart: {{ include "mariadb-galera.chart" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+spec:
+  selector:
+    matchLabels:
+      app.kubernetes.io/instance: {{ .Release.Name }}
+      app.kubernetes.io/name: {{ include "mariadb-galera.name" . }}
+  serviceName: {{ template "mariadb-galera.fullname" . }}
+  replicas: {{ .Values.replicas }}
+  updateStrategy:
+    type: {{ .Values.updateStrategy.type }}
+    {{- if (eq "Recreate" .Values.updateStrategy.type) }}
+    rollingUpdate: null
+    {{- end }}
+  template:
+    metadata:
+      {{- if .Values.annotations }}
+      annotations:
+        {{- range $key, $value := .Values.annotations }}
+        {{ $key }}: '{{ $value }}'
+        {{- end }}
+      {{- end }}
+      labels:
+        app.kubernetes.io/name: {{ include "mariadb-galera.name" . }}
+        app.kubernetes.io/instance: {{ .Release.Name }}
+        helm.sh/chart: {{ include "mariadb-galera.chart" . }}
+    spec:
+      {{- if .Values.schedulerName }}
+      schedulerName: {{ .Values.schedulerName | quote }}
+      {{- end }}
+      serviceAccountName: {{ template "mariadb-galera.serviceAccountName" . }}
+      {{- if .Values.securityContext.enabled }}
+      securityContext:
+        fsGroup: {{ .Values.securityContext.fsGroup }}
+        runAsUser: {{ .Values.securityContext.runAsUser }}
+      {{- end }}
+      {{- if eq .Values.antiAffinity "hard" }}
+      affinity:
+      {{- with .Values.affinity  }}
+{{ toYaml . | indent 8 }}
+      {{- end }}
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            - topologyKey: "kubernetes.io/hostname"
+              labelSelector:
+                matchLabels:
+                  app.kubernetes.io/name: {{ include "mariadb-galera.name" . }}
+                  app.kubernetes.io/instance: {{ .Release.Name }}
+      {{- else if eq .Values.antiAffinity "soft" }}
+      affinity:
+      {{- with .Values.affinity  }}
+{{ toYaml . | indent 8 }}
+      {{- end }}
+        podAntiAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+          - weight: 1
+            podAffinityTerm:
+              topologyKey: kubernetes.io/hostname
+              labelSelector:
+                matchLabels:
+                  app.kubernetes.io/name: {{ include "mariadb-galera.name" . }}
+                  app.kubernetes.io/instance: {{ .Release.Name }}
+      {{- else}}
+      {{- with .Values.affinity }}
+      affinity:
+{{ toYaml . | indent 8 }}
+      {{- end }}
+      {{- end }}
```

Comment:
Suggestion for affinity (using templates):
```yaml
      affinity:
        {{- with .Values.nodeAffinity }}
        nodeAffinity: {{ tpl (toYaml .) $ | nindent 10 }}
        {{- end }}
        {{- with .Values.podAffinity }}
        podAffinity: {{ tpl (toYaml .) $ | nindent 10 }}
        {{- end }}
        {{- if eq .Values.podAntiAffinity "hard" }}
        podAntiAffinity:
          requiredDuringSchedulingIgnoredDuringExecution:
          - topologyKey: "kubernetes.io/hostname"
            labelSelector:
              matchLabels:
                app.kubernetes.io/name: {{ include "mariadb-galera.name" . }}
                app.kubernetes.io/instance: {{ .Release.Name }}
        {{- else if eq .Values.podAntiAffinity "soft" }}
        podAntiAffinity:
          preferredDuringSchedulingIgnoredDuringExecution:
          - weight: 1
            podAffinityTerm:
              topologyKey: "kubernetes.io/hostname"
              labelSelector:
                matchLabels:
                  app.kubernetes.io/name: {{ include "mariadb-galera.name" . }}
                  app.kubernetes.io/instance: {{ .Release.Name }}
        {{- end }}
```

Reply:
:+1: